contracts/BaseStrategy.sol

```diff
@@ -258,9 +256,33 @@ abstract contract BaseStrategy {
      * would be negatively affected if `harvest()` is not called shortly, then this can return `true`
      * even if the keeper might be "at a loss" (keepers are always reimbursed by yEarn)
      *
+     * NOTE: `callCost` must be priced in terms of `want`
+     *
      * NOTE: this call and `tendTrigger` should never return `true` at the same time.
      */
-    function harvestTrigger(uint256 gasCost) public virtual view returns (bool);
+    function harvestTrigger(uint256 callCost) public view returns (bool) {
+        StrategyParams memory params = vault.strategies(address(this));
+
+        // Should not trigger if strategy is not activated
+        if (params.activation == 0) return false;
+
+        // If some amount is owed, pay it back
+        // NOTE: Since debt is adjusted in step-wise fashion, it is appropiate to always trigger here,
+        //       because the resulting change should be large (might not always be the case)
+        uint256 outstanding = vault.debtOutstanding();
+        if (outstanding > 0) return true;
```

Comment:
This needs some more thought. If a strategy is slowly withdrawing - for instance in a timelock situation. we don't want harvest being called every block.

Reply:
This will never happen, because (and the comment reflects this) debt changes occur in a step-wise fashion. So basically if you have 100k and then your debt limit drops to 50k, you will have a debt outstanding of 50k and should try to unwind enough to pay that back

Reply:
Debt is not always adjusted stepwise. For instance in an emergency shutdown situation. But even if it is strategy may not be able to free up 50k. But I suppose those edge cases can rewrite the whole method. Might be worth leaving virtual in the function declaration, that gives strategist the option to override if they have an edge case. Like you've done with tend trigger.

Reply:
fixed in cb2efcf